service/matching/matching_engine.go

```diff
@@ -917,6 +923,73 @@ func (e *matchingEngineImpl) getDefaultBuildId(tqMgr taskQueuePartitionManager)
 	return "", nil
 }
 
+func (e *matchingEngineImpl) getBuildIdTaskReachability(
+	ctx context.Context,
+	nsID,
+	nsName,
+	taskQueue,
+	buildId string,
+) (enumspb.BuildIdTaskReachability, error) {
+	getResp, err := e.GetWorkerVersioningRules(ctx, &matchingservice.GetWorkerVersioningRulesRequest{
```

Comment:
we are already in the root partition which is the owner of the user data. so we don't have to make a call through matching_engine. instead we can ask the root tq partition manager locally. that way you don't have to add deleted_rule_inclusion_period to the proto.

Reply:
addressed